Detectors/CTP/workflowLumi/src/RawToLumiConverterSpec.cxx

```diff
@@ -0,0 +1,176 @@
+  for (auto it = parser.begin(); it != parser.end(); ++it) {
+    auto rdh = it.get_if<o2::header::RAWDataHeader>();
+    auto triggerOrbit = o2::raw::RDHUtils::getTriggerOrbit(rdh);
+
+    auto feeID = o2::raw::RDHUtils::getFEEID(rdh); // 0 = IR, 1 = TCR
+    auto linkCRU = (feeID & 0xf00) >> 8;
+    if (linkCRU == o2::ctp::GBTLinkIDIntRec) {
+      payloadCTP = o2::ctp::NIntRecPayload;
+    } else {
+      continue;
+    }
+    if (first) {
+      orbit0 = triggerOrbit;
+      first = false;
+    }
+    LOG(debug) << "RDH FEEid: " << feeID << " CTP CRU link:" << linkCRU << " Orbit:" << triggerOrbit;
+    // TF in 128 bits words
+    gsl::span<const uint8_t> payload(it.data(), it.size());
+    gbtword80_t gbtWord = 0;
+    int wordCount = 0;
+    std::vector<gbtword80_t> diglets;
+    if (orbit0 != triggerOrbit) {
+      // create lumi per HB
+      lumiPoint lp;
+      lp.counts = countsMB;
+      lp.ir.orbit = triggerOrbit;
+      lumiPointsHBF1.push_back(lp);
+      // LOG(info) << "Orbit:" << triggerOrbit << " tvx count:" << countsMB;
+      countsMB = 0;
+      //
+      remnant = 0;
+      size_gbt = 0;
+      orbit0 = triggerOrbit;
+    }
+    for (auto payloadWord : payload) {
+      // LOG(info) << wordCount << " payload:" <<  int(payloadWord);
+      if (wordCount == 15) {
+        wordCount = 0;
+      } else if (wordCount > 9) {
+        wordCount++;
+      } else if (wordCount == 9) {
+        for (int i = 0; i < 8; i++) {
+          gbtWord[wordCount * 8 + i] = bool(int(payloadWord) & (1 << i));
+        }
+        wordCount++;
+        diglets.clear();
+        // LOG(info) << " gbtword:" << gbtWord;
+        o2::ctp::reco_workflow::RawToDigitConverterSpec::makeGBTWordInverse(diglets, gbtWord, remnant, size_gbt, payloadCTP);
+        // count tvx
+        for (auto diglet : diglets) {
+          // LOG(info) << " diglet:" << diglet;
+          gbtword80_t pld = (diglet & mTVXMask);
+          if (pld.count() != 0) {
+            countsMB++;
+          }
+        }
+        gbtWord = 0;
+      } else {
+        // std::cout << "wordCount:" << wordCount << std::endl;
+        for (int i = 0; i < 8; i++) {
+          gbtWord[wordCount * 8 + i] = bool(int(payloadWord) & (1 << i));
+        }
+        wordCount++;
+      }
+    }
+  }
+  // LOG(info) << "lumiPoints size:" << lumiPointsHBF1.size();
+  mOutputLumiPoints.clear();
+  float_t countHBFn = 0;
+  for (int ihb = 0; ihb < lumiPointsHBF1.size(); ihb++) {
+    countHBFn += lumiPointsHBF1[ihb].counts;
+    lumiPoint lp;
+    if (ihb < mHBFsToAverage) {
+      lp.nHBFs = ihb + 1;
+      lp.ir = lumiPointsHBF1[ihb].ir;
+    } else {
+      countHBFn -= lumiPointsHBF1[ihb - mHBFsToAverage].counts;
+      lp.nHBFs = mHBFsToAverage;
+      lp.ir = lumiPointsHBF1[ihb - mHBFsToAverage + 1].ir;
+    }
+    lp.counts = countHBFn;
+    // LOG(info) << "lp orbit:" << lp.ir.orbit << " lp.countes:" << lp.counts << " T:" << lp.nHBFs;
+    mOutputLumiPoints.push_back(lp);
+  }
```

Comment:
Not sure this is amounts for ~1s running average we need: the accumulation is done only within 1 TF. I would simply define a data members
```
float mCounts = 0.;
size_t mNTFToIntegrate = 0;
size_t mNHBIntegrated = 0;
std::deque<float> mHistory;
```
and the number of TFs to integrate as an option (instead of the NHB) and assign it as:
```
float tfCounts = 0; // counts in this TF
for (int ihb = 0; ihb < lumiPointsHBF1.size(); ihb++) {
  tfCounts += lumiPointsHBF1[ihb].counts;
}
mHistory.push_back(tfCounts);
mCounts += tfCounts;
if (mHistory.size() < mNTFToIntegrate) { // still did not reach target NTFs
  mNHBIntegrated += lumiPointsHBF1.size(); // Assuming that we always have the same size in all TFs
} else {
  mCounts -= mHistory.front();
  mHistory.pop_front();
}
// normalize
float meanLumi = mNHBIntegrated ? mCounts / (mNHBIntegrated*o2::contacts::lhc::LHCOrbitMUS*1e-6) : 0;
// send meanLumi
```

Reply:
@lietava  What do you mean by `how many mCounts do we need then`? We should request only counting period, above I put `mNTFToIntegrate` for that. I assume that lumiPointsHBF1.size() is the same for all TFs and each entry always corresponds to 1 orbit. 
For the return value, indeed better to return a simple struct like
```
struct LumiInfo {
  size_t mCounts = 0;
  size_t mNHBFCounted = 0; 
};
```